src/components/FilterBar.jsx

```diff
@@ -40,13 +40,27 @@ export default function FilterBar(props) {
   const {
     changeSortBy,
     changeSortFormat,
-    onAmountFilterChange,
+    setQueryParam,
+    clearQueryParam,
     queryParams,
   } = props;
   const classes = useStyles();
 
-  const minAmount = queryParams.minAmount ?? 0;
-  const maxAmount = queryParams.maxAmount ?? 0;
+  const { minAmount, maxAmount } = queryParams;
+
+  if (
+    minAmount !== undefined &&
+    !(Number.isInteger(minAmount) && minAmount > 0)
+  ) {
+    clearQueryParam('minAmount');
+  }
+
+  if (
+    maxAmount !== undefined &&
+    !(Number.isInteger(maxAmount) && maxAmount > 0)
+  ) {
+    clearQueryParam('maxAmount');
+  }
```

Comment:
I'd keep this check in Scholarships.jsx so that the effect fetching scholarships doesn't run multiple times in these cases.

Reply:
For sure, I removed the pruneQueryParam from FilterBar.jsx since it wasn't being used.